ddl/index.go

```diff
@@ -1814,46 +1817,48 @@ func (w *worker) addTableIndex(t table.Table, reorgInfo *reorgInfo) error {
 			if err != nil {
 				break
 			}
-			finish, err = w.updateReorgInfo(tbl, reorgInfo)
+			finish, err = updateReorgInfo(w.sessPool, tbl, reorgInfo)
 			if err != nil {
 				return errors.Trace(err)
 			}
 		}
 	} else {
 		//nolint:forcetypeassert
 		phyTbl := t.(table.PhysicalTable)
-		// TODO: Support typeAddIndexMergeTmpWorker.
-		if reorgInfo.Job.ReorgMeta.IsDistReorg && !reorgInfo.mergingTmpIdx {
-			sCtx, err := w.sessPool.get()
-			if err != nil {
-				return errors.Trace(err)
-			}
-			defer w.sessPool.put(sCtx)
-			return w.controlWritePhysicalTableRecord(newSession(sCtx), phyTbl, typeAddIndexWorker, reorgInfo)
-		}
 		err = w.addPhysicalTableIndex(phyTbl, reorgInfo)
 	}
 	return errors.Trace(err)
 }
 
-// updateReorgInfo will find the next partition according to current reorgInfo.
-// If no more partitions, or table t is not a partitioned table, returns true to
-// indicate that the reorganize work is finished.
-func (w *worker) updateReorgInfo(t table.PartitionedTable, reorg *reorgInfo) (bool, error) {
+func getPartitionRangeKey(reorg *reorgInfo, t table.PartitionedTable, pid int64) (kv.Key, kv.Key, error) {
+	if reorg.mergingTmpIdx {
+		indexID := reorg.currElement.ID
+		startKey, endKey := tablecodec.GetTableIndexKeyRange(pid, tablecodec.TempIndexPrefix|indexID)
+		return startKey, endKey, nil
+	}
+
+	currentVer, err := getValidCurrentVersion(reorg.d.store)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	return getTableRange(reorg.d.jobContext(reorg.Job.ID), reorg.d, t.GetPartition(pid), currentVer.Ver, reorg.Job.Priority)
+}
+
+func getNextPartitionInfo(reorg *reorgInfo, t table.PartitionedTable, currPhysicalTableID int64) (int64, kv.Key, kv.Key, error) {
 	pi := t.Meta().GetPartitionInfo()
 	if pi == nil {
-		return true, nil
+		return 0, nil, nil, nil
```

Comment:
if 0 is a special value? could we define it to a const?

Reply:
The fact that tableID has a value of 0 is itself illegal